### DISCUSSION TITLE
Question regarding `--no-back-referencing`

### DISCUSSION THREAD

**Discussion by emaincourt** [aarondl/sqlboiler#1116] [Q&A] [answered]
Hi,

I have a question regarding the usage of the `--no-back-referencing` flag. For most of the code, disabling back referencing seems to work pretty nicely. However, after taking a look at the templates, I noticed [this](https://github.com/volatiletech/sqlboiler/blob/c6e262b8478df41a168168eab221d3a477b1607c/templates/main/12_relationship_to_many_setops.go.tpl#L147) in `12_relationship_to_many_setops.go.tpl`:

```
	{{if .ToJoinTable -}}
	for _, rel := range related {
		if rel.R == nil {
			rel.R = &{{$ftable.DownSingular}}R{
				{{$relAlias.Foreign}}: {{$ltable.UpSingular}}Slice{{"{"}}o{{"}"}},
			}
		} else {
			rel.R.{{$relAlias.Foreign}} = append(rel.R.{{$relAlias.Foreign}}, o)
		}
	}
	{{else -}}
	for _, rel := range related {
		if rel.R == nil {
			rel.R = &{{$ftable.DownSingular}}R{
				{{$relAlias.Foreign}}: o,
			}
		} else {
			rel.R.{{$relAlias.Foreign}} = o
		}
	}
	{{end -}}
```

Basically to summarize the behaviour, after adding the relationship, it assigns the current object as a relation to the newly attached one. Resulting in pretty much a back reference. Actually the documentation of the function itself describes the behaviour correctly:

```
// Add{{$relAlias.Local}} adds the given related objects to the existing relationships
// of the {{$table.Name | singular}}, optionally inserting them as new records.
// Appends related to o.R.{{$relAlias.Local}}.
// Sets related.R.{{$relAlias.Foreign}} appropriately. <- This line
```

However, in case the back referencing is disabled, shouldn't it ignore the assignment step to `rel.R` ? Are there any motivations for referencing it ?

I would be very pleased to help fixing if needed of course.

Thanks in advance for your answer.

**Comment by stephenafamo** [comment, marked as answer]
Thanks for pointing this out. It has been fixed in `v4.9.1`.

**Comment by emaincourt** [comment]
Thanks a lot @stephenafamo. That was fast.
